### PR TITLE
feat: migrate FormattedNumber to ts

### PR DESCRIPTION
**Pull request by indremak**
Migrated following files to TS (#910 ):

- [x] FormattedNumber.jsx
- [x] FormattedNumberConsts.js
- [x] FormattedNumberHelpers.js
- [x] FormattedNumber.jest.js

### CODE REVIEW

File: src/components/FormattedNumber/FormattedNumber.tsx [hunk 1]
```diff
@@ -1,13 +1,67 @@
-import React, { useMemo, forwardRef } from "react";
-import PropTypes from "prop-types";
+import React, { useMemo, forwardRef, ForwardedRef } from "react";
 import classNames from "classnames";
 import { formatNumber, formatNumberConsts } from "../../helpers/textManipulations";
 import { baseClassName } from "./FormattedNumberConsts";
 import { validateValue } from "./FormattedNumberHelpers";
 import "./FormattedNumber.scss";
+import VibeComponentProps from "src/types/VibeComponentProps";
```
Comment (orrgottlieb) [inline src/components/FormattedNumber/FormattedNumber.tsx:7]: great job, due to internal requirements please use the relative path 🙏

File: src/components/FormattedNumber/FormattedNumber.tsx [hunk 2]
```diff
@@ -1,13 +1,67 @@
...
+import VibeComponent from "src/types/VibeComponent";
```
Comment (orrgottlieb) [inline src/components/FormattedNumber/FormattedNumber.tsx:8]: same comment as well 🙏

**Review by orrgottlieb — CHANGES_REQUESTED** [review]
Awesome job, please see my 2 small comments

File: src/components/FormattedNumber/FormattedNumber.tsx (hunk 1 above)
Comment (indremak) [reply src/components/FormattedNumber/FormattedNumber.tsx]: Thanks for review. I've updated both imports to use relative path.

File: src/components/FormattedNumber/FormattedNumber.tsx (hunk 2 above)
Comment (indremak) [reply src/components/FormattedNumber/FormattedNumber.tsx]: Fixed

**Review by orrgottlieb — APPROVED** [review]

**Comment by orrgottlieb** [comment]
@indremak thank you so much for this awesome PR!
will merge it and close the issue with hacktoberfest-approved 🙏

**Pull request merged**